lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -390,6 +394,38 @@ func (handler *ContainersHandlersImpl) ContainerWaitHandler(params containers.Co
 	}
 }
 
+func (handler *ContainersHandlersImpl) RenameContainerHandler(params containers.ContainerRenameParams) middleware.Responder {
+	defer trace.End(trace.Begin(fmt.Sprintf("Rename container to %s", params.Name)))
+
+	h := exec.GetHandle(params.Handle)
+	if h == nil || h.ExecConfig == nil {
+		return containers.NewContainerRenameNotFound()
+	}
+
+	// get the indicated container for rename
+	container := exec.Containers.Container(h.ExecConfig.ID)
+	if container == nil {
+		return containers.NewContainerRenameNotFound()
+	}
+
+	if container.ExecConfig.Name == params.Name {
+		return containers.NewContainerRenameConflict()
```

Comment:
I think you need another error type for this failure. name conflict usually means the new name is conflict with other containers, but if it's same to itself, usually returns different message.
You can compare it with docker command